doc/guides/prog_guide/cnet.rst

```diff
@@ -0,0 +1,400 @@
+
+The **ch_callback** is the function pointer to the application function defined in the *channel* API. The
+callback function is called for receiving data and helping inform the application about connections coming
+and going. The primary reason for callback is receiving packet data needs to be handled in the thread context
+as the stack instance to help eliminate the need for some types of locking.
+
+The **ch_node** is the node associated with the transmit channel structure or the next node to allow the
+application thread to enqueue packet data into a given CNET graph instance. The **ch_rcv** and **ch_snd** are
+used to receive and send data to/from the application.
+
+The :ref:`CNET Chnl Buffer structure <figure_cnet_chnl_buffer>` is part of the Chnl structure and manages the
+channel data.
+
+.. _figure_cnet_chnl_buffer:
+
+.. code-block:: c
+   :caption: CNET Channel Buffer layout
+
+       struct chnl_buf {
+           pktmbuf_t **cb_vec;     /**< Vector of mbuf pointers */
+           uint32_t cb_cc;         /**< actual chars in buffer */
+           uint32_t cb_hiwat;      /**< high water mark */
+           uint32_t cb_lowat;      /**< low water mark */
+           uint32_t cb_size;       /**< protocol send/receive size */
+       };
+
+Protocol Control Block (PCB)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The *PCB* structure contains information about the protocol connection and which network interface is connected
+to the given instance of a connection. THe *PCB* is used by the protocol packet handling to locate an active
+connection as in a listening connection.
+
+.. note:: Add more information and info about the *PCB* structure below the structure.
+
+.. _figure_pcb_structure:
+
+.. code-block:: c
+   :caption: CNET Protocol Control Block
+
+       struct pcb_key {
+           struct in_caddr faddr; /**< foreign IP address */
+           struct in_caddr laddr; /**< local IP address */
+       } __cne_aligned(sizeof(void *));
+
+       struct pcb_entry {
+           TAILQ_ENTRY(pcb_entry) next; /**< Pointer to the next pcb_entry in a list */
+           struct pcb_key key;          /**< Key values for PCB entry */
+           struct netif *netif;         /**< Netif pointer */
+           struct chnl *ch;             /**< Channel pointer */
+           struct tcb_entry *tcb;       /**< TCB pointer */
+           uint16_t opt_flag;           /**< Option flags */
+           uint8_t ttl;                 /**< Time to live */
+           uint8_t tos;                 /**< TOS value */
+           uint8_t closed;              /**< Closed flag */
+           uint8_t ip_proto;            /**< IP protocol number */
+       } __cne_cache_aligned;
+
+The **pcb_key** structure defines the local and foreign addresses (note: it currently only handles IPv4
+addresses), which define a connection and is how the connection is found by the protocol processing.
+The **next** structure is a link list of *PCBs* attached to *half open* or *backlog* queues for
```

Comment:
link list -> linked list... though I'm beginning to think you intentionally wrote 'link list' since its used in mulitple places.

Reply:
When I get something in my head I tend to put it everywhere, it should be linked list.